source/slang/slang-check-decl.cpp

```diff
@@ -9308,8 +9308,160 @@ namespace Slang
             this, funcDecl, attr, DeclAssociationKind::PrimalSubstituteFunc);
     }
 
+    static int32_t _findLevelsOfInheritance(AggTypeDecl* decl, int32_t inheritanceLevel, List<AggTypeDecl*>& linearInheritanceHierarchy)
+    {
+        // We need to assume multiple `InheritanceDecl` are possible per `decl` member list.
+        int32_t inheritanceDepth = inheritanceLevel;
+        for (auto maybeInheritance : decl->members)
+        {
+            auto inheritanceDecl = as<InheritanceDecl>(maybeInheritance);
+            if (!inheritanceDecl)
+                continue;
+            auto validInheritanceBase = as<DeclRefType>(inheritanceDecl->base.type);
+            if (!validInheritanceBase || !validInheritanceBase->getDeclRef())
+                continue;
+            auto validInheritanceBaseType = as<AggTypeDecl>(validInheritanceBase->getDeclRef().getDecl());
+            if (!validInheritanceBaseType)
+                continue;
+            linearInheritanceHierarchy.add(validInheritanceBaseType);
+            inheritanceDepth = std::max(inheritanceDepth, _findLevelsOfInheritance(validInheritanceBaseType, inheritanceLevel+1, linearInheritanceHierarchy));
+        }
+        return inheritanceDepth;
+    }
+
+    static void _findAndLegalizeAllCtorforStructFieldInitExpression(SemanticsDeclAttributesVisitor* visitor, ASTBuilder* m_astBuilder, StructDecl* structDecl, List<ConstructorDecl*>& ctorList, int32_t levelsOfInheritance)
+    {
+        ConstructorDecl* ctorDefault = nullptr;
+        for (auto maybeCtor : structDecl->members)
+        {
+            auto ctor = as<ConstructorDecl>(maybeCtor);
+            if (!ctor)
+                continue;
+            ctorList.add(ctor);
+
+            // It is possible BlockStmt has a child with the type of
+            // `ExpressionStmt` if an existing constructor has only 1
+            // expression. This would be a senario we need to
+            // put the `ExpressionStmt` inside a `SeqStmt`.
+            // Since we found an init expression, we need to ensure
+            // we are adding to a constructor which can handle multiple
+            // statments since we will be adding `init` expressions to
+            // the constructor.
```

Comment:
Please make this sentence more readable.
I don't think I understood fully but it could be something like,
```
If a constructor can handle multiple statements, "init" is added to the constructor.
```